crates/http/src/backend/hyper/mod.rs

```diff
@@ -0,0 +1,224 @@
+use std::fmt::Debug;
+use std::net::SocketAddr;
+
+use scuffle_context::ContextFutExt;
+#[cfg(feature = "tracing")]
+use tracing::Instrument;
+
+use crate::error::Error;
+use crate::service::{HttpService, HttpServiceFactory};
+
+mod handler;
+mod stream;
+mod utils;
+
+/// A backend that handles incoming HTTP connections using a hyper backend.
+///
+/// This is used internally by the [`HttpServer`](crate::server::HttpServer) but can be used directly if preferred.
+///
+/// Call [`run`](HyperBackend::run) to start the server.
+#[derive(Debug, Clone, bon::Builder)]
+pub struct HyperBackend<F> {
```

Comment:
Likewise with the h3 builder we should add comments to each field to display them in the docs